lsp-methods.el

```diff
@@ -855,16 +973,39 @@ directory."
       (puthash root lsp--cur-workspace lsp--workspaces)
       (lsp-mode 1)
       (run-hooks 'lsp-before-initialize-hook)
-      (setq init-params
-            `(:processId ,(emacs-pid)
+
+      (setf
+       extra-init-params-resolved (if (functionp extra-init-params)
+                                      (funcall extra-init-params lsp--cur-workspace)
+                                    extra-init-params)
+       ;; join the saved workspace folders + the folders comming from the
+       ;; language extension to keep backward compatibility with `lsp-java'
+       workspace-folders (append
+                          (lsp--read-from-file
+                           (concat (file-name-as-directory root) ".folders"))
```

Comment:
Is it conventional to use `$root/.folders`?

Reply:
I am open to alternative suggestions.

Reply:
@yyoncho You might think I can merge PRs in this repo but I can't.. The most I can give you is an approval..